### PR TITLE
security.User.getUsersWithPermissions: Add includeInactive flag

### PR DESCRIPTION
**Pull request by labkey-alan**
#### Rationale
See rationale in the [platform PR](https://github.com/LabKey/platform/pull/4870).

#### Related Pull Requests
* https://github.com/LabKey/platform/pull/4870
* https://github.com/LabKey/labkey-api-js/pull/164
* https://github.com/LabKey/labkey-ui-components/pull/1328
* https://github.com/LabKey/labkey-ui-premium/pull/235
* https://github.com/LabKey/biologics/pull/2475
* https://github.com/LabKey/inventory/pull/1076
* https://github.com/LabKey/sampleManagement/pull/2191

#### Changes
* security.User.getUsersWithPermissions: Add includeInactive flag

### CODE REVIEW

File: src/labkey/security/User.ts [hunk 1]
```diff
@@ -179,6 +179,11 @@ export interface GetUsersOptions extends RequestCallbackOptions<GetUsersResponse
      * only users whose email address or display name starts with the value supplied will be returned.
      */
     name?: string;
+    /**
+     * This value is used to optionally include deactivated members in server responses. Only used with
+     * getUsersWithPermissions.
+     */
+    includeDeactivated?: boolean;
```
Comment (labkey-nicka) [inline src/labkey/security/User.ts]: Move this property to the `interface GetUsersWithPermissionsOptions` (line 206).

File: src/labkey/security/User.ts [hunk 2]
```diff
@@ -179,6 +179,11 @@ export interface GetUsersOptions extends RequestCallbackOptions<GetUsersResponse
      * only users whose email address or display name starts with the value supplied will be returned.
      */
     name?: string;
+    /**
```
Comment (labkey-nicka) [inline src/labkey/security/User.ts]:
```suggestion
    /**
     * Include deactivated members in server responses. Does not apply when "groups" is specified. Defaults to false.
     */
```

File: src/labkey/security/User.ts [hunk 3]
```diff
@@ -253,6 +258,8 @@ function getUsersRequest(endpoint: string, config: GetUsersOptions): XMLHttpRequ
         } else {
             params.permissions = [config.permissions];
         }
+
+        params.includeDeactivated = !!config.includeDeactivated;
```
Comment (labkey-nicka) [inline src/labkey/security/User.ts]: Consider not specifying this at all when not defined.

File: src/labkey/security/User.ts (hunk 1 above)
Comment (labkey-alan) [reply src/labkey/security/User.ts]: In order to do that I had to get rid of the shared getUsersRequest method.

**Review by labkey-nicka — APPROVED** [review]

File: src/labkey/security/User.ts (hunk 1 above)
Comment (labkey-nicka) [reply src/labkey/security/User.ts]: Nicely refactored.

**Pull request merged**